src/config/frontend/EdFi.DmsConfigurationService.Frontend.AspNetCore/Modules/ApplicationModule.cs

```diff
@@ -3,12 +3,199 @@
+        {
+            ApplicationQueryResult.Success success => Results.Ok(success.ApplicationResponses),
+            ApplicationQueryResult.FailureUnknown failure => Results.Problem(statusCode: 500),
+            _ => Results.Problem(statusCode: 500),
+        };
+    }
+
+    private static async Task<IResult> GetById(
+        long id,
+        [FromServices] HttpContext httpContext,
+        [FromServices] IApplicationRepository applicationRepository,
+        [FromServices] ILogger<ApplicationModule> logger
+    )
+    {
+        ApplicationGetResult getResult = await applicationRepository.GetApplication(id);
+        return getResult switch
+        {
+            ApplicationGetResult.Success success => Results.Ok(success.ApplicationResponse),
+            ApplicationGetResult.FailureNotFound => Results.NotFound(),
+            ApplicationGetResult.FailureUnknown => Results.Problem(statusCode: 500),
+            _ => Results.Problem(statusCode: 500),
+        };
+    }
+
+    private static async Task<IResult> Update(
+        long id,
+        [FromServices] ApplicationUpdateCommandValidator validator,
+        [FromBody] ApplicationUpdateCommand command,
+        [FromServices] HttpContext httpContext,
+        [FromServices] IApplicationRepository repository
+    )
+    {
+        await validator.GuardAsync(command);
+
+        var vendorUpdateResult = await repository.UpdateApplication(command);
+
+        if (vendorUpdateResult is ApplicationUpdateResult.FailureVendorNotFound)
+        {
+            throw new ValidationException(
+                new[] { new ValidationFailure("VendorId", $"Reference 'VendorId' does not exist.") }
+            );
+        }
+
+        return vendorUpdateResult switch
+        {
+            ApplicationUpdateResult.Success success => Results.NoContent(),
+            ApplicationUpdateResult.FailureNotExists => Results.NotFound(),
+            ApplicationUpdateResult.FailureUnknown => Results.Problem(statusCode: 500),
+            _ => Results.Problem(statusCode: 500),
+        };
+    }
+
+    private static async Task<IResult> Delete(
+        long id,
+        [FromServices] HttpContext httpContext,
+        [FromServices] IApplicationRepository repository,
+        [FromServices] IClientRepository clientRepository,
+        [FromServices] ILogger<ApplicationModule> logger
+    )
+    {
+        logger.LogInformation("Deleting Application {id}", id);
+
+        var apiClientsResult = await repository.GetApplicationApiClients(id);
+        switch (apiClientsResult)
+        {
+            case ApplicationApiClientsResult.Success success:
+                foreach (var clientUuid in success.ClientUuids)
+                {
+                    try
+                    {
+                        logger.LogInformation("Deleting client {clientUuid}", clientUuid);
+                        bool deleted = await clientRepository.DeleteClientAsync(clientUuid.ToString());
+                        if (!deleted)
```

Comment:
While deleting, I see we are deleting the apiclient from the Keycloak. Should we delete the record from the dmscs.ApiClient table as well ?

Reply:
Delete cascades are on at the database level, so they will be deleted.